tensorflow/lite/experimental/shlo/ops/convolution_test.cc

```diff
@@ -0,0 +1,615 @@
+                                          .element_type = DataType::kSI64},
+                       .data = window_stride_values.data()};
+  Tensor padding{.type = TensorType{.shape = shape_padding,
+                                    .element_type = DataType::kSI64},
+                 .data = padding_values.data()};
+  Tensor lhs_dilation{.type = TensorType{.shape = shape_parametrs,
+                                         .element_type = DataType::kSI64},
+                      .data = lhs_dilation_values.data()};
+  Tensor rhs_dilation{.type = TensorType{.shape = shape_parametrs,
+                                         .element_type = DataType::kSI64},
+                      .data = rhs_dilation_values.data()};
+  Tensor window_reversal{.type = TensorType{.shape = shape_parametrs,
+                                            .element_type = DataType::kI1},
+                         .data = window_reversal_values.data()};
+  Tensor inputSpatialDimensions{
+      .type =
+          TensorType{.shape = shape_parametrs, .element_type = DataType::kSI64},
+      .data = inputSpatialDimensions_values.data()};
+  Tensor kernel_spatial_dimensions{
+      .type =
+          TensorType{.shape = shape_parametrs, .element_type = DataType::kSI64},
+      .data = kernel_spatial_dimensions_values.data()};
+  Tensor output_spatial_dimensions{
+      .type =
+          TensorType{.shape = shape_parametrs, .element_type = DataType::kSI64},
+      .data = output_spatial_dimensions_values.data()};
+  Tensor output_tensor{
+      .type = QuantizedPerAxisTensorType{.shape = shape_result,
+                                         .element_type = tensor_type_axis_res},
+      .data = output_data.data()};
+
+  absl::InlinedVector<PrecisionTypes, 2> precision_configs = {
+      PrecisionTypes::DEFAULT, PrecisionTypes::DEFAULT};
+  auto op = Create(ConvolutionOp::Attributes{
+      .window_strides = window_stride,
+      .padding = padding,
+      .lhs_dilation = lhs_dilation,
+      .rhs_dilation = rhs_dilation,
+      .window_reversal = window_reversal,
+      .input_batch_dimension = input_batch_dimension,
+      .input_feature_dimension = input_feature_dimension,
+      .input_spacial_dimensions = inputSpatialDimensions,
+      .kernel_input_feature_dimension = kernel_input_feature_dimension,
+      .kernel_output_feature_dimension = kernel_output_feature_dimension,
+      .kernel_spacial_dimensions = kernel_spatial_dimensions,
+      .output_batch_dimension = output_batch_dimension,
+      .output_feature_dimension = output_feature_dimension,
+      .output_spacial_dimensions = output_spatial_dimensions,
+      .feature_group_count = feature_group_count,
+      .batch_group_count = batch_group_count,
+      .precision_configs = precision_configs});
+
+  Vector<StorageT> expected_data =
+      Vector<StorageT>{5, 10, 15, 20, 25, 5, 10, 15, 20, -25};
+  ;
+  if (std::is_same<StorageT, I4>::value) {
```

Comment:
I'd suggest splitting out a separate test case for i4 where we can ensure the input and expected data are within the i4 range without having to clamp most of the values.

Reply:
resolved by calling quantize function ,now works for all int types without clamping